From 4f1c1c624cb52fb19520f3ae8cd6008517d84daf Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Sun, 10 May 2026 23:41:21 +0800
Subject: [PATCH 091/102] RUYI: riscv: dts: eswin: add Pine64 StarPro64 board
 device tree

Add minimal device tree for the Pine64 StarPro64 board based on the
ESWIN EIC7700 SoC. Currently the data populated in this DT file is
for UART.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 arch/riscv/boot/dts/eswin/Makefile            |  1 +
 .../dts/eswin/eic7700-pine64-starpro64.dts    | 29 +++++++++++++++++++
 2 files changed, 30 insertions(+)
 create mode 100644 arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts

diff --git a/arch/riscv/boot/dts/eswin/Makefile b/arch/riscv/boot/dts/eswin/Makefile
index 5daade822e15..26ff4314d0b4 100644
--- a/arch/riscv/boot/dts/eswin/Makefile
+++ b/arch/riscv/boot/dts/eswin/Makefile
@@ -1,3 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0
 dtb-$(CONFIG_ARCH_ESWIN) += eic7700-hifive-premier-p550.dtb
 dtb-$(CONFIG_ARCH_ESWIN) += eic7700-milkv-megrez.dtb
+dtb-$(CONFIG_ARCH_ESWIN) += eic7700-pine64-starpro64.dtb
diff --git a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
new file mode 100644
index 000000000000..9780d802e61d
--- /dev/null
+++ b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright (c) 2024, Beijing ESWIN Computing Technology Co., Ltd.
+ */
+
+/dts-v1/;
+
+#include "eic7700.dtsi"
+
+/ {
+	compatible = "pine64,starpro64", "eswin,eic7700";
+	model = "Pine64 StarPro64";
+
+	aliases {
+		serial0 = &uart0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&uart2 {
+	status = "okay";
+};
-- 
2.47.3

